### PR TITLE
Edns0options

### LINKED ISSUE #46: parse and generate EDNS0 options
Better support for "popular" EDNS0 options from the [IANA list](https://www.iana.org/assignments/dns-parameters/dns-parameters.xhtml#dns-parameters-11).  I've the start of a patch for ECS that allows this:

```js
    options: [ {
      code: 8,
      ip: 'fe80::/64'
    } ]
```

on parse, you get this:

```js
       options:
        [ { code: 8,
            data: <Buffer 00 02 40 00 fe 80 00 00 00 00 00 00>,
            family: 2,
            sourcePrefixLength: 64,
            scopePrefixLength: 0,
            ip: 'fe80::' } ] } ] }
```

note that the data field is retained for backward compatibility.

If this approach makes sense and is interesting to the maintainers, I'll send a PR eventually with support for all of the options in the IANA list that have RFCs that are clear enough to make sense to me, including tests.

### PR DESCRIPTION
**Pull request by hildjj**
Fixes #46.  Adds support for several relevant EDNS0 types.  Updates tests.  Updates README.

### CODE REVIEW

**Comment by silverwind** [comment]
What do you think about these? 1:1 mapping of code to name only:

````js
case 'OPTION_0': return 0
case 'LLQ': return 1
case 'UL': return 2
case 'NSID': return 3
case 'OPTION_4': return 4
case 'DAU': return 5
case 'DHU': return 6
case 'N3U': return 7
case 'CLIENT-SUBNET': return 8
case 'EXPIRE': return 9
case 'COOKIE': return 10
case 'TCP-KEEPALIVE': return 11
case 'PADDING': return 12
case 'CHAIN': return 13
case 'KEY-TAG': return 14
case 'DEVICEID': return 26946
case 'OPTION_65535': return 65535
````

Again, if you like to use `ECS` for option 8, I'm also happy to use it. I don't like ambiguity so I think we should not take multiple names for the same option.

File: test.js [hunk 1]
```diff
@@ -509,6 +535,40 @@ tape('unpack', function (t) {
...
+    const s = optioncodes.toString(code)
+    t.ok(compare(t, s, str), `${code} => ${str}`)
+    t.ok(compare(t, optioncodes.toCode(s), code), `${str} => ${code}`)
+    for (const a of aliases) {
+      t.ok(compare(t, optioncodes.toCode(a), code), `${a} => ${code}`)
+    }
+  }
+  t.end()
+})
+
```
Comment (silverwind) [inline test.js]: While this is exemplary, I think it wil ultimately only makes adding support for new option codes harder while having not much testing value in itself. I'd remove it.

**Comment by silverwind** [comment]
Generally looking good to me, except the naming thing above.

File: test.js (hunk 1 above)
Comment (hildjj) [reply test.js]: Remove the whole test or just the aliases bit?

Comment (silverwind) [reply test.js]: Meant the whole test, but on second thought, it's probably ok to gain some coverage, let's keep it in :wink:

**Pull request merged**